zend_abstract_interface/methods/php5/methods.c

```diff
@@ -0,0 +1,113 @@
+#include "../methods.h"
+
+#include <Zend/zend_interfaces.h>
+#include <assert.h>
+#include <sandbox/sandbox.h>
+
+zend_class_entry *zai_class_lookup_ex(const char *cname, size_t cname_len TSRMLS_DC) {
+    if (!cname || !cname_len) return NULL;
+    zend_class_entry **ce;
+    /* Since we do not want to invoke the autoloader and we assume the caller
+     * will pass in the lowercased class name, we look up the class entry from
+     * the EG(class_table) directly versus calling zend_lookup_class_ex().
+     */
+    return (zend_hash_find(EG(class_table), cname, (cname_len + 1), (void **)&ce) == SUCCESS) ? *ce : NULL;
+}
+
+#ifndef NDEBUG
+static bool z_is_lower(const char *str) {
+    char *p = (char *)str;
+    while (*p) {
+        if (isalpha(*p) && !islower(*p)) return false;
+        p++;
+    }
+    return true;
+}
+#endif
+
+static bool z_call_method_without_args_ex(zval *object, zend_class_entry *ce, const char *method, size_t method_len,
+                                          zval **retval TSRMLS_DC) {
+    if (!ce || !method || !method_len || !retval) return false;
+
+    /* Prevent a potential dangling pointer in case the caller accidentally
+     * sent in an allocated retval.
+     */
+    if (*retval != NULL) return false;
+
+    /* If 'object' is NULL, the caller wants to call this method statically. */
+    zval **obj = object ? &object : NULL;
+
+    /* This one gets me all the time. Trying to save myself 5 minutes for the
+     * next time it inevitably happens.
+     */
+    assert(z_is_lower(method) && "Don't forget to send those method names in all lowercase. ;)");
```

Comment:
We should add the same check in `zai_class_lookup_ex`, I think? It requires an lcname.

Reply:
Indeed! Good point. I included this assertion in class lookups. I also added an assertion for root-scope prefixes. (62c9b73)